src/website/app/demos/Menu/bestPractices.js

```diff
@@ -0,0 +1,286 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import React from 'react';
+import { createStyledComponent, pxToEm } from '../../../../utils';
+import Menu, { MenuDivider, MenuGroup, MenuItem } from '../../../../Menu';
+
+const DemoLayout = createStyledComponent('div', ({ theme }) => ({
+  backgroundColor: theme.color_gray_10,
+  padding: theme.spacing_double,
+
+  '& > div': {
+    backgroundColor: 'white',
+    width: pxToEm(224)
+  }
+}));
+
+export default {
+  menu: [
+    {
+      type: 'do',
+      title: 'use the appropriate variant for your intent',
+      example: (
+        <DemoLayout>
+          <Menu>
+            <MenuItem>Share</MenuItem>
+            <MenuItem>Edit</MenuItem>
+            <MenuItem>Duplicate</MenuItem>
+            <MenuItem variant="danger">Delete</MenuItem>
+          </Menu>
+        </DemoLayout>
+      ),
+      description: `Match Mineral UI's pre-defined variants with semantic purpose.`
+    },
+    {
+      type: 'do',
+      title: 'use clear labeling for option messaging',
+      example: (
+        <DemoLayout>
+          <Menu>
+            <MenuItem secondaryText="Cmd + L">Select Line</MenuItem>
+            <MenuItem secondaryText="⌃ + ⇧ + W">Select Word</MenuItem>
+            <MenuItem secondaryText="Cmd + J">Join Lines</MenuItem>
+          </Menu>
+        </DemoLayout>
+      ),
+      description: `[MenuItem](../menu-item) actions should be predictable for frictionless interaction.
+  Labels should be structured: \`<verb> <noun>\`.`
+    },
+    {
+      type: 'dont',
+      title: 'cause confusion with the wrong variant',
+      example: (
+        <DemoLayout>
+          <Menu>
+            <MenuItem>Create New Table</MenuItem>
+            <MenuItem>Clone Table</MenuItem>
+            <MenuItem variant="success">Drop Table</MenuItem>
+          </Menu>
+        </DemoLayout>
+      ),
+      description:
+        'Using the wrong variant can cause confusion, or undue stress for users.'
+    },
+    {
+      type: 'dont',
+      title: 'use vague option labeling',
+      example: (
+        <DemoLayout>
+          <Menu>
+            <MenuItem>Go</MenuItem>
+            <MenuItem>Various Config Options</MenuItem>
+            <MenuItem>Do It</MenuItem>
+          </Menu>
+        </DemoLayout>
+      ),
+      description:
+        'Users should know exactly what will happen when they choose an option.'
+    }
+  ],
+  menuDivider: [
+    {
+      type: 'do',
+      title: 'group elements to establish hierarchy',
```

Comment:
Using "group" here is perhaps unclear, since we also have MenuGroup. What about:

"group elements" ➡️  "divide menu options" ?

Reply:
I think another, possibly more common, use case is setting off a meaningfully different menu option, e.g. a menu with a bunch of CRUD-like actions, and setting off 'Delete' from the rest.

Reply:
updated. I added another "do" example with the CRUD example, good thinking.